RNMT-3078 Upgrade PushWoosh

EiyuuZack opened the pull request:
## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- Finishing by placing the link to the issue here preceded by either 'Closes' OR 'Fixes' -->
In order to overcome a security vulnerability detected on Android, PushWoosh is being upgraded to a version higher than 7.16.0.

References https://outsystemsrd.atlassian.net/browse/RNMT-3078

## Type of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply -->
- [x] Feature (change which adds functionality)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [ ] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Components affected
- [x] Android
- [x] iOS

## Tests
<!--- Describe how you tested your changes in detail -->
<!--- Include details of your test environment if relevant -->
Manual tests were performed following the PushWoosh test plan which will not be linked here.

## Checklist
<!--- Go over all the following items and put an `x` in all the boxes that apply -->
- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [ ] Code follows code style of this project
- [ ] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.14.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.14.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.14.1

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.14.1

Commit:
[PUSH-20076] fix build problem occuring when use config-file tag inside config.xml

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.15.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.15.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.15.1

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.15.1

Commit:
[PUSH-20527] fix build issue

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.15.2

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.15.2

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.16.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.16.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.17.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.17.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.17.1

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.17.1

EiyuuZack committed:
Merge branch 'master' into feat/RNMT-3078/upgrade-mabs5

usernuno commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pushwoosh-cordova-plugin",
-  "version": "7.13.0-OS1",
+  "version": "7.17.1",
```
Should keep the -OS suffix

Chuckytuh replied:
@EiyuuZack, @usernuno is this just an upgrade from upstream without changes on our end? because if it is, and if I recall correctly, then the suffix isn't needed. The suffix is added only when our branch/repo deviates from upstream.

EiyuuZack replied:
It contains our changes to plugin.xml to lock Android support libs to API 26

usernuno commented on plugin.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<plugin xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android" id="pushwoosh-cordova-plugin" version="7.13.0-OS1">
+<plugin xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android" id="pushwoosh-cordova-plugin" version="7.17.1">
```
Should keep the -OS suffix

usernuno commented on plugin.xml:
```diff
@@ -60,8 +60,8 @@
-		<framework src="com.google.firebase:firebase-core:16.0.4" />
-		<framework src="com.google.firebase:firebase-messaging:17.3.4" />
+		<framework src="com.google.firebase:firebase-core:(+,17.0.0)" />
+		<framework src="com.google.firebase:firebase-messaging:(+,19.0.0)" />
```
Is this safe? This can cause AndroidX related issues (I assume it does not since the app compiled in order to do tests).

But it can cause conflicts with the Firebase Plugin from the Experts, we already had a support case with this issue. Please make sure both plugins can coexist.

pmaroco replied:
@EiyuuZack don't forget to check this!

EiyuuZack replied:
Indeed there were conflicts with Firebase Plugin. Question here is whether we should be the ones to accommodate an unsupported plugin or they should comply with the service.

In the interest of time, because it is a given this will not be addressed in the short term, the dependencies were downgraded to match Firebase's.

usernuno replied:
I agree, we just did change that to close a high priority support case.

usernuno reviewed (changes requested):
Please check the feedback I provided.

EiyuuZack committed:
chore: add suffix to version

The whole point of this issue is to upgrade the plugin from upstream
therefore it makes sense to update the version before reaching the
outsystems branch.

References https://outsystemsrd.atlassian.net/browse/RNMT-3078

EiyuuZack committed:
build: downgrade firebase dependencies

Match those specified by https://github.com/TruewindIT/cordova-plugin-firebase/blob/1.6/plugin.xml

References https://outsystemsrd.atlassian.net/browse/RNMT-3078

(cherry picked from commit b431416931fc0041a0730b856af7b321942e730e)
